lib/moesif_aws_lambda/moesif_aws_middleware.rb

```diff
@@ -68,7 +91,27 @@ def @moesif_helpers.log_debug(message)
       end
     end
 
-    def parse_body(body, headers)
+    def process_body(body_wrapper, headers)
+      # the `event` object can either be a Hash or a String
+      if body_wrapper.instance_of?(Hash) and body_wrapper.key?("body")
+        body = body_wrapper.fetch("body")
+        begin
+          if body_wrapper.fetch("isBase64Encoded", false) and is_base64_str(body)
+            return body, "base64"
+          else
+            return safe_json_parse(body, headers)
+          end
+          rescue
+            return base64_encode_body(body)
+        end
+      elsif body_wrapper.instance_of?(String)
+        return base64_encode_body(body_wrapper)
+      else
+        return nil, nil
```

Comment:
I think we'd call safe_json_parse here because what if the body_wrapper is an instance_of?(Array) then also we'd want to try/parse as json

Reply:
@keyur9 I found [this](https://docs.aws.amazon.com/lambda/latest/dg/ruby-handler.html#ruby-handler-basics:~:text=The%20two%20objects%20that%20the%20handler%20accepts%20are%20the%20invocation%20event%20and%20context.%20The%20event%20is%20a%20Ruby%20object%20that%20contains%20the%20payload%20that%27s%20provided%20by%20the%20invoker.%20If%20the%20payload%20is%20a%20JSON%20document%2C%20the%20event%20object%20is%20a%20Ruby%20hash.%20Otherwise%2C%20it%27s%20a%20string.) in AWS docs; so I thought skipping this edge case.

> The two objects that the handler accepts are the invocation event and context. The event is a Ruby object that contains the payload that's provided by the invoker. If the payload is a JSON document, the event object is a Ruby hash. Otherwise, it's a string.

Should we still check?